libkbfs/conflict_resolver.go

```diff
@@ -2331,8 +2337,11 @@ func (cr *ConflictResolver) doActions(ctx context.Context,
 		// Now get the directory blocks.  For unmerged directories, we
 		// can use a nil local block cache, because unmerged blocks
 		// should never be changed during the CR process (since
-		// they're just going away).
-		unmergedDir := cr.fbo.blocks.newDirDataWithLBC(
+		// they're just going away).  This call will lock `blockLock`,
+		// and the subsequent `newDirData` calls can assume it's
+		// locked already.
+		var unmergedDir *dirData
+		unmergedDir, undoFn = cr.fbo.blocks.newDirDataWithLBC(
```

Comment:
The "undoFn" here doesn't actually undo what'd done by `newDirDataWithLBC`. It just releases the lock. So perhaps it should be called something else? (here and other occurrences of `undoFn` in this PR)

Reply:
@strib normally in a similar situation, you'd create a `func()` inside the loop, with a `defer undoFn` once the function is assigned. This time you've chosen to go with the nillable method variable with several defers. Any particular reason why?

Reply:
@jzila: the span of the subsequent code seemed too complicated to put into an anonymous `func()`, but I can try to split it into a new, real function on `cr` instead.  It'll cause conflicts with other PRs I think, but I can manage that.

@songgao: I was just trying to encapsulate the logic a bit -- I didn't want the caller to really be aware of the locking so I picked something generic.  But I'll try to come up with a better name.

Reply:
@strib you can always implement that as a later PR to avoid conflicts.